app/controllers/tasks_controller.rb

```diff
@@ -1,12 +1,14 @@
 # frozen_string_literal: true
 
 class TasksController < ApplicationController
+  before_action :authenticate_user!
   before_action :set_task, only: %i[show edit update destroy]
 
   # GET /tasks
   # GET /tasks.json
   def index
-    @tasks = Task.all
+    is_completed = params[:completed].present?
```

Comment:
I noticed, if I send `tasks?completed=false` as params to this route I will also see completed tasks. There's no way to do this via the UI. Did you considered any other approaches to solve the filters problem?